targets/TARGET_STM/us_ticker_32b.c

```diff
@@ -23,33 +23,30 @@
 
 TIM_HandleTypeDef TimMasterHandle;
 
-static int us_ticker_inited = 0;
-
 void us_ticker_init(void)
 {
-    if (us_ticker_inited) return;
-    us_ticker_inited = 1;
-
-    TimMasterHandle.Instance = TIM_MST;
-
-    HAL_InitTick(0); // The passed value is not used
+    /* NOTE: assuming that HAL tick has already been initialized! */
 }
 
 uint32_t us_ticker_read()
 {
-    if (!us_ticker_inited) us_ticker_init();
     return TIM_MST->CNT;
 }
 
 void us_ticker_set_interrupt(timestamp_t timestamp)
 {
-    TimMasterHandle.Instance = TIM_MST;
+    /* Disable global IRQs */
+    core_util_critical_section_enter();
```

Comment:
@betzw There is no need to add synchronization mechanism at that level. It is acknowledge that the function `us_ticker_set_interrupt` is not thread safe and not irq safe. 

However, synchronization is handled by the generic ticker API whenever events are [inserted](https://github.com/ARMmbed/mbed-os/blob/master/hal/mbed_ticker_api.c#L195) or [removed](https://github.com/ARMmbed/mbed-os/blob/master/hal/mbed_ticker_api.c#L235).